File: camerax_android/android/src/main/kotlin/dev/hebei/camerax_android/legacy/ml/Detector.kt
package dev.hebei.camerax_android.legacy.ml

import dev.hebei.camerax_android.legacy.CameraXRegistrar
import dev.hebei.camerax_android.legacy.PigeonApiDetector

class Detector(registrar: CameraXRegistrar) : PigeonApiDetector(registrar) {
    abstract class Stub internal constructor(internal open val obj: com.google.mlkit.vision.interfaces.Detector<*>)
}

File: camerax/lib/src/ml/detector.dart
import 'package:camerax_platform_interface/camerax_platform_interface.dart'
    as $interface;
import 'package:meta/meta.dart';

abstract class Detector<T> {
  @internal
  $interface.Detector<T> get obj;

  Future<void> close() => obj.close();
}
